Better error needs what providing a wrong argument to the class constructor

```python
src/ewoksppf/tests/test_ppf_workflow25.py:71: in <module>
    class Good(Task, inputs_names=["reason"], output_names=["result", "reason"]):
../ewokscore/src/ewokscore/task.py:189: in __init_subclass__
    super().__init_subclass__(**kwargs)
../ewokscore/src/ewokscore/registration.py:11: in __init_subclass__
    super().__init_subclass__(**kwargs)
../ewokscore/src/ewokscore/hashing.py:190: in __init_subclass__
    super().__init_subclass__(**kwargs)
E   TypeError: Good.__init_subclass__() takes no keyword arguments
```

typo in `inputs_names`

--- a/src/ewokscore/task.py
+++ b/src/ewokscore/task.py
@@ -5,6 +5,7 @@ import re
 import time
 from contextlib import ExitStack
 from contextlib import contextmanager
+from difflib import get_close_matches
 from typing import Any
 from typing import Generator
 from typing import Mapping
@@ -186,6 +187,32 @@ class Task(Registered, UniversalHashable, register=False):
         output_model: Union[Type[BaseInputModel], None] = None,
         **kwargs,
     ):
+        # Hard code the parameters to __init_subclass__ to check for misspellings
+        task_param_names = {
+            "input_names",
+            "optional_input_names",
+            "output_names",
+            "n_required_positional_inputs",
+            "input_model",
+            "output_model",
+        }
+
+        # Use difflib to find potential typos in kwargs
+        bad_kwargs = {}
+        for kwarg_name in kwargs.keys():
+            matches = get_close_matches(kwarg_name, task_param_names, n=1, cutoff=0.6)
+            if matches:
+                bad_kwargs[kwarg_name] = matches[0]
+
+        if bad_kwargs:
+            suggestion_msgs = []
+            for invalid, suggested in bad_kwargs.items():
+                suggestion_msgs.append(f"Did you mean '{suggested}'? (got '{invalid}')")
+            raise TypeError(
+                f"{subclass.__name__}.__init_subclass__() got invalid keyword argument(s): "
+                f"{', '.join(sorted(bad_kwargs.keys()))}. " + ". ".join(suggestion_msgs)
+            )
+
         super().__init_subclass__(**kwargs)
 
         input_names_set, optional_input_names_set = subclass._generate_inputs_sets(

--- a/src/ewokscore/tests/test_task.py
+++ b/src/ewokscore/tests/test_task.py
@@ -148,6 +148,27 @@ def test_task_required_positional_inputs():
         MyTask()
 
 
+def test_init_subclass_rejects_typos_with_suggestions():
+    with pytest.raises(TypeError) as exc_info:
+
+        class Bad(Task, input=["reason"], output_name=["result", "reason"]):
+            pass
+
+    # checks exception for input_names is caught
+    assert "input_names" in str(exc_info.value)
+    assert "input" in str(exc_info.value)
+
+    # checks exception for output_names is caught
+    assert "output_names" in str(exc_info.value)
+    assert "output_name" in str(exc_info.value)
+
+
+def test_init_subclass_accepts_valid_params():
+    # Should not raise an error
+    class Good(Task, input_names=["reason"], output_names=["result", "reason"]):
+        pass
+
+
 def test_task_cleanup_references():
     class MyTask(Task, input_names=["mylist"], output_names=["mylist"]):
         def run(self):

--- a/src/ewokscore/tests/test_task.py
+++ b/src/ewokscore/tests/test_task.py
@@ -1,5 +1,6 @@
 import gc
 import json
+import re
 from glob import glob
 from pathlib import Path
 
@@ -148,19 +149,46 @@ def test_task_required_positional_inputs():
         MyTask()
 
 
-def test_init_subclass_rejects_typos_with_suggestions():
-    with pytest.raises(TypeError) as exc_info:
+def test_init_subclass_rejects_input_name_typo():
+    # test that the wrong name appears in the error statement
+    with pytest.raises(TypeError, match=re.escape("input")):
 
-        class Bad(Task, input=["reason"], output_name=["result", "reason"]):
+        class Bad1(Task, input=["reason"]):
             pass
 
-    # checks exception for input_names is caught
-    assert "input_names" in str(exc_info.value)
-    assert "input" in str(exc_info.value)
+    # test that the correct name appears somewhere in the error statement
+    with pytest.raises(TypeError, match=re.escape("input_names")):
 
-    # checks exception for output_names is caught
-    assert "output_names" in str(exc_info.value)
-    assert "output_name" in str(exc_info.value)
+        class Bad2(Task, input=["reason"]):
+            pass
+
+
+def test_init_subclass_rejects_optional_input_names_typo():
+    # test that the wrong name appears in the error statement
+    with pytest.raises(TypeError, match=re.escape("optinal_input_names")):
+
+        class Bad1(Task, optinal_input_names=["reason"]):
+            pass
+
+    # test that the correct name appears somewhere in the error statement
+    with pytest.raises(TypeError, match=re.escape("optinal_input_names")):
+
+        class Bad2(Task, optinal_input_names=["reason"]):
+            pass
+
+
+def test_init_subclass_rejects_output_name_typo():
+    # test that the wrong name appears in the error statement
+    with pytest.raises(TypeError, match="output_name"):
+
+        class Bad1(Task, output_name=["reason"]):
+            pass
+
+    # test that the correct name appears in the error statement
+    with pytest.raises(TypeError, match="output_names"):
+
+        class Bad2(Task, output_name=["reason"]):
+            pass
 
 
 def test_init_subclass_accepts_valid_params():

